### PR TITLE
Added etcd storage

### PR DESCRIPTION
**Pull request by g4s8**
For #305 - introduced etcd storage, added integration test
using junit extension.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -21,10 +21,18 @@ which requires to implement the following functionalities:
 * list files in a directory
 * check if a file/directory exists
 
-Supported abstractions:
-
-- [x] [File system](https://www.javadoc.io/doc/com.artipie/asto/latest/com/artipie/asto/fs/FileStorage.html) 
-- [X] S3
+Dictionary used for ASTO:
+ - `Storage` - key-value based storage
+ - `Key` - storage keys, could be conferted to strings
```
Comment (olenagerasimova) [inline README.md:26]: @g4s8 seems like typo `conferted` -> `converted`

File: src/main/java/com/artipie/asto/etcd/EtcdStorage.java [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+            return new CompletableFutureSupport.Failed<Void>(
+                new IllegalStateException(
+                    String.format("Content size must be in range (1;%d)", EtcdStorage.MAX_SIZE)
+                )
+            ).get();
+        }
+        return new Concatenation(content).single()
+            .map(buf -> new Remaining(buf).bytes())
+            .map(bytes -> ByteSequence.from(bytes))
+            .to(SingleInterop.get())
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/asto/etcd/EtcdStorage.java:117]: @g4s8 you can simply use `PublisherAs()` here to read all bytes from `content`

File: src/main/java/com/artipie/asto/etcd/EtcdStorage.java [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+        ).thenApply(
+            kv -> kv.orElseThrow(
+                () -> new KeyNotFoundException(key)
+            ).getValue().getBytes()
+        ).thenApply(bytes -> new Content.From(bytes));
+    }
+
+    @Override
+    public CompletableFuture<Void> delete(final Key key) {
+        return this.client.getKVClient().delete(keyToSeq(key)).thenApply(ignore -> (Void) null);
```
Comment (olenagerasimova) [inline src/main/java/com/artipie/asto/etcd/EtcdStorage.java:157]: @g4s8 hm, will this method fail if key does not exist? It should according to `Storage` contract

**Review by olenagerasimova — CHANGES_REQUESTED** [review]
@g4s8 thanks, please, check my comments

**Review by olenagerasimova — APPROVED** [review]
@g4s8 thanks

**Comment by g4s8** [comment]
@olenagerasimova thanks

**Comment by 0pdd** [comment]
@g4s8 2 puzzles [#309](https://github.com/artipie/asto/issues/309), [#310](https://github.com/artipie/asto/issues/310) are still not solved.

**Comment by 0pdd** [comment]
@g4s8 2 puzzles [#310](https://github.com/artipie/asto/issues/310), [#384](https://github.com/artipie/asto/issues/384) are still not solved; solved: [#309](https://github.com/artipie/asto/issues/309).

**Comment by 0pdd** [comment]
@g4s8 the puzzle [#384](https://github.com/artipie/asto/issues/384) is still not solved; solved: [#309](https://github.com/artipie/asto/issues/309), [#310](https://github.com/artipie/asto/issues/310).

**Pull request merged**